# HG changeset patch
# User andrew
# Date 1421870534 0
#      Wed Jan 21 20:02:14 2015 +0000
# Node ID bbff2a7acd335d23a7946a1b7eedd42946946311
# Parent  7fc691671bf07fe00940e02452235f711e50112e
OPENJDK6-51: Sync patch for 4873188 with 7 version
Reviewed-by: omajid

diff -r 7fc691671bf0 -r bbff2a7acd33 src/share/classes/sun/security/ssl/CipherBox.java
--- openjdk/jdk/src/share/classes/sun/security/ssl/CipherBox.java	Tue Jan 20 03:06:20 2015 +0000
+++ openjdk/jdk/src/share/classes/sun/security/ssl/CipherBox.java	Wed Jan 21 20:02:14 2015 +0000
@@ -500,10 +500,9 @@
                     byte[] buf = null;
                     int limit = bb.limit();
                     if (bb.hasArray()) {
-                        int arrayOffset = bb.arrayOffset();
                         buf = bb.array();
-                        System.arraycopy(buf, arrayOffset + pos + blockSize,
-                            buf, arrayOffset + pos, limit - pos - blockSize);
+                        System.arraycopy(buf, pos + blockSize,
+                                         buf, pos, limit - pos - blockSize);
                         bb.limit(limit - blockSize);
                     } else {
                         buf = new byte[limit - pos - blockSize];
diff -r 7fc691671bf0 -r bbff2a7acd33 src/share/classes/sun/security/ssl/SSLEngineImpl.java
--- openjdk/jdk/src/share/classes/sun/security/ssl/SSLEngineImpl.java	Tue Jan 20 03:06:20 2015 +0000
+++ openjdk/jdk/src/share/classes/sun/security/ssl/SSLEngineImpl.java	Wed Jan 21 20:02:14 2015 +0000
@@ -1261,6 +1261,14 @@
                 writer.writeRecord(eor, ea, writeMAC, writeCipher);
 
         /*
+         * turn off the flag of the first application record if we really
+         * consumed at least byte.
+         */
+        if (isFirstAppOutputRecord && ea.deltaApp() > 0) {
+            isFirstAppOutputRecord = false;
+        }
+
+        /*
          * We only need to check the sequence number state for
          * non-handshaking record.
          *
@@ -1279,14 +1287,6 @@
             }
         }
 
-        /*
-         * turn off the flag of the first application record if we really
-         * consumed at least byte.
-         */
-        if (isFirstAppOutputRecord && ea.deltaApp() > 0) {
-            isFirstAppOutputRecord = false;
-        }
-
         return hsStatus;
     }
 
